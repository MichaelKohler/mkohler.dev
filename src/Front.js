import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  faBook, faHeart,
} from '@fortawesome/free-solid-svg-icons';

const Front = () => {
  return (
    <div className="content-wrap">
      <section id="homesection" className="container-fluid nopadding">
        <section className="m-details row nopadding bg-color01">
          <div className="col-md-8 nopadding">
            <div className="padding-50">
              <div className="row nopadding">
                <div className="col-md-12 nopadding">
                  <h2 className="font-accident-two-normal uppercase">About Michael</h2>
                  <div className="quote">
                    <h5 className="font-accident-one-bold uppercase subtitle">
                      JavaScript frontend and backend developer with a passion for Web Standards
                    </h5>
                  </div>
                </div>
              </div>

              <section className="row infoblock-container">
                <div className="col-md-5 infoblock">
                  <div className="row">
                    <div className="icon-container col-sm-1 col-md-3 hidden-sm hidden-xs">
                      <FontAwesomeIcon icon={faBook} />
                    </div>
                    <div className="col-sm-11 col-md-9">
                      <h5 className="font-accident-one-bold uppercase">Eager to learn</h5>
                      <p>
                        Even though I have several years of experience with Software Development
                        I&apos;m always eager to learn more to increase my knowledge.
                      </p>
                    </div>
                  </div>
                </div>
                <div className="col-md-1" />
                <div className="col-md-5 infoblock nopadding">
                  <div className="row">
                    <div className="icon-container col-sm-1 col-md-3 hidden-sm hidden-xs">
                      <FontAwesomeIcon icon={faHeart} />
                    </div>
                    <div className="col-sm-11 col-md-9">
                      <h5 className="font-accident-one-bold uppercase">Motivated</h5>
                      <p>
                        With my volunteer work for Mozilla I&apos;m very passionate about the Open
                        Web and therefore very motivated to work as Web Developer.
                      </p>
                    </div>
                  </div>
                </div>
              </section>
            </div>
          </div>

          <section className="col-md-4 personal nopadding ui-block-color01">
            <div className="padding-50">
              <h3 className="font-accident-two-normal uppercase">Personal Info</h3>
              <div className="personal-info fullwidth">
                <div className="row">
                  <div className="col-lg-5 col-sm-12">Name:</div>
                  <div className="col-lg-7 col-sm-12">Michael Kohler</div>
                </div>
                <div className="row">
                  <div className="col-lg-5 col-sm-12">Date of Birth:</div>
                  <div className="col-lg-7 col-sm-12">30/11/1991</div>
                </div>
                <div className="row">
                  <div className="col-lg-5 col-sm-12">Place:</div>
                  <div className="col-lg-7 col-sm-12">
                    Berlin, Germany
                    <br />
                    originally from Bern, Switzerland
                  </div>
                </div>
                <div className="row">
                  <div className="col-lg-5 col-sm-12">eMail:</div>
                  <div className="col-lg-7 col-sm-12">
                    <a href="mailto:me@michaelkohler.info">me@michaelkohler.info</a>
                  </div>
                </div>
                <div className="row">
                  <div className="col-lg-5 col-sm-12">Languages (fluent):</div>
                  <div className="col-lg-7 col-sm-12">German, English</div>
                </div>
              </div>
            </div>
          </section>
        </section>

        <section className="row nopadding ui-block-color02">
          <div className="col-md-4 pro-experience nopadding">
            <div className="padding-50">
              <h3 className="font-accident-two-normal uppercase">Experience</h3>
              <div className="experience">
                <ul className="">
                  <li className="date">04/2008 - now</li>
                  <li className="company uppercase">
                    <a href="https://www.mozilla.org">Volunteer @ Mozilla Foundation</a>
                  </li>
                  <li className="position">Developer, Community Builder</li>
                </ul>
                <ul className="">
                  <li className="date">02/2019 - now</li>
                  <li className="company uppercase">
                    <a href="http://www.control4.com">Control4</a>
                  </li>
                  <li className="position">(acquired NEEO)</li>
                </ul>
                <ul className="">
                  <li className="date">11/2016 - 01/2019</li>
                  <li className="company uppercase">
                    <a href="http://www.neeo.com">NEEO AG</a>
                  </li>
                  <li className="position">JavaScript Developer</li>
                </ul>
                <ul className="">
                  <li className="date">04/2015 - 10/2016</li>
                  <li className="company uppercase">
                    <a href="http://www.netcase.ch">NETCASE GmbH</a>
                  </li>
                  <li className="position">Web Developer, NodeJS and AngularJS</li>
                </ul>
                <ul className="">
                  <li className="date">01/2014 - 02/2015</li>
                  <li className="company uppercase">
                    <a href="http://www.swisscom.ch">Swisscom Banking Provider AG</a>
                  </li>
                  <li className="position">
                    Application Developer, 2nd-Level-Support Document Output
                    Management, Requirement Engineering
                  </li>
                </ul>
                <ul className="">
                  <li className="date">08/2010 - 12/2013</li>
                  <li className="company uppercase">
                    <a href="http://www.entris-banking.ch">Entris Banking AG</a>
                  </li>
                  <li className="position">
                    Application Developer, 2nd-Level-Support
                    Document Output Management, Requirement Engineering
                  </li>
                </ul>
              </div>

              <h3 className="font-accident-two-normal uppercase">Education</h3>
              <div className="experience">
                <ul className="">
                  <li className="date">02/2013 - 02/2017</li>
                  <li className="company uppercase">
                    <a href="https://www.ffhs.ch">Fernfachhochschule Schweiz</a>
                  </li>
                  <li className="position">Computer Science, not finished</li>
                </ul>
                <ul className="">
                  <li className="date">09/2011 - 01/2013</li>
                  <li className="company uppercase">
                    <a href="http://www.inf.unibe.ch/index_eng.html">University of Bern</a>
                  </li>
                  <li className="position">Computer Science, not finished</li>
                </ul>
                <ul className="">
                  <li className="date">08/2006 - 07/2010</li>
                  <li className="company uppercase">
                    <a href="http://www.gymneufeld.ch">Gymnasium Neufeld</a>
                  </li>
                  <li className="position">Matura</li>
                </ul>
              </div>
            </div>
          </div>

          <div className="col-md-8 circle-skills nopadding ui-block-color03">
            <div className="padding-50">
              <h3 className="font-accident-two-normal uppercase">My Professional skills</h3>
              <div className="row">
                <div className="col-sm-4 single-skill">
                  <h4 className="font-accident-one-normal uppercase">Backend JavaScript</h4>
                  <p className="font-regular-normal">
                    I&apos;ve been developing Node.js applications for the past 8 years, as private
                    projects as well as professionally. At NEEO I developped JavaScript
                    for both front and back end based on Ionic and Node.js in a IoT environment.
                    Previously during my time at NETCASE GmbH I had also worked on Sails.js /
                    Loopback based APIs. Currently at Control4 I&apos;m developping Node.js
                    applications.
                  </p>
                </div>
                <div className="col-sm-4 single-skill">
                  <h4 className="font-accident-one-normal uppercase">Frontend JavaScript</h4>
                  <p className="font-regular-normal">
                    As part of my work at NEEO and NETCASE GmbH and several private projects
                    I used AngularJS 1 for the past 4 years. From really simple to
                    more complex frontends, I&apos;ve been involved in developing them. While my
                    CSS skills are not perfect, I know AngularJS/React well and can output
                    semantic DOM to make it as easy as possible to style. In the recent past
                    I&apos;ve mostly used React for my personal as well as Mozilla projects.
                  </p>
                </div>
                <div className="col-sm-4 single-skill">
                  <h4 className="font-accident-one-normal uppercase">Other languages</h4>
                  <p className="font-regular-normal">
                    I have experience in several other languages, most notably HTML and CSS.
                    Additionally I have also worked with TypeScript. I
                    haven&apos;t written any PHP, Ruby or Java in years, but back in the days...
                    Sometime in the future I will also learn Lua and Rust.
                  </p>
                </div>
              </div>
              <div className="row">
                <div className="col-sm-4 single-skill">
                  <h4 className="font-accident-one-normal uppercase">Web Standards</h4>
                  <p className="font-regular-normal">
                    Due to my affiliation with Mozilla, I&apos;m really passionate about the Open
                    Web. Therefore I&apos;m trying to keep on top of changes to Web Standards and
                    will always encourage the use of modern standards where possible.
                  </p>
                </div>
                <div className="col-sm-4 single-skill">
                  <h4 className="font-accident-one-normal uppercase">DevOps</h4>
                  <p className="font-regular-normal">
                    At NEEO I also gained experience in DevOps. I like working with Docker
                    containers and try to make sure Infrastructure as Code is applied where
                    possible. For my personal servers (even my Raspberry Pi) I&apos;ve been
                    using Ansible. Currently I&apos;m looking into Kubernetes.
                  </p>
                </div>
                <div className="col-sm-4 single-skill">
                  <h4 className="font-accident-one-normal uppercase">Databases</h4>
                  <p className="font-regular-normal">
                    I&apos;ve been mostly using MongoDB as database. Nevertheless I have used
                    several other database systems in the past, such as MySQL.
                  </p>
                </div>
              </div>
            </div>
          </div>
        </section>
      </section>
    </div>
  );
};

export default Front;
