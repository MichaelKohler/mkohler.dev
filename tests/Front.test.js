import React from 'react';
import { shallow } from 'enzyme';

import Front from '../src/Front';

test('should correctly render Front', () => {
  const component = shallow(<Front />);
  expect(component).toMatchSnapshot();
});
