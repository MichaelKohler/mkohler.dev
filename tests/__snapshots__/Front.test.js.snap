// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render Front 1`] = `
<section
  id="homesection"
>
  <section
    className="m-details row bg-color01"
  >
    <div
      className="col-md-12 col-lg-8 full-height-container"
    >
      <div
        className="row"
      >
        <div
          className="col-md-12"
        >
          <h2>
            About Michael
          </h2>
          <h3
            className="quote"
          >
            JavaScript developer with a passion for Web Standards and DevOps
          </h3>
        </div>
      </div>
      <section
        className="row infoblock-container"
      >
        <div
          className="col-md-5 infoblock"
        >
          <div
            className="row"
          >
            <div
              className="icon-container col-sm-1 col-md-3 d-none d-lg-block"
            >
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon={
                  Object {
                    "icon": Array [
                      448,
                      512,
                      Array [],
                      "f02d",
                      "M448 360V24c0-13.3-10.7-24-24-24H96C43 0 0 43 0 96v320c0 53 43 96 96 96h328c13.3 0 24-10.7 24-24v-16c0-7.5-3.5-14.3-8.9-18.7-4.2-15.4-4.2-59.3 0-74.7 5.4-4.3 8.9-11.1 8.9-18.6zM128 134c0-3.3 2.7-6 6-6h212c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H134c-3.3 0-6-2.7-6-6v-20zm0 64c0-3.3 2.7-6 6-6h212c3.3 0 6 2.7 6 6v20c0 3.3-2.7 6-6 6H134c-3.3 0-6-2.7-6-6v-20zm253.4 250H96c-17.7 0-32-14.3-32-32 0-17.6 14.4-32 32-32h285.4c-1.9 17.1-1.9 46.9 0 64z",
                    ],
                    "iconName": "book",
                    "prefix": "fas",
                  }
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                swapOpacity={false}
                symbol={false}
                title=""
                transform={null}
              />
            </div>
            <div
              className="col-sm-11 col-md-9"
            >
              <h4>
                Eager to learn
              </h4>
              <p>
                Even though I have several years of experience with Software Development I'm always eager to learn more to increase my knowledge.
              </p>
            </div>
          </div>
        </div>
        <div
          className="col-md-1"
        />
        <div
          className="col-md-5 infoblock"
        >
          <div
            className="row"
          >
            <div
              className="icon-container col-sm-1 col-md-3 d-none d-lg-block"
            >
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon={
                  Object {
                    "icon": Array [
                      512,
                      512,
                      Array [],
                      "f004",
                      "M462.3 62.6C407.5 15.9 326 24.3 275.7 76.2L256 96.5l-19.7-20.3C186.1 24.3 104.5 15.9 49.7 62.6c-62.8 53.6-66.1 149.8-9.9 207.9l193.5 199.8c12.5 12.9 32.8 12.9 45.3 0l193.5-199.8c56.3-58.1 53-154.3-9.8-207.9z",
                    ],
                    "iconName": "heart",
                    "prefix": "fas",
                  }
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                swapOpacity={false}
                symbol={false}
                title=""
                transform={null}
              />
            </div>
            <div
              className="col-sm-11 col-md-9"
            >
              <h4>
                Motivated
              </h4>
              <p>
                With my volunteer work for Mozilla I'm very passionate about the Open Web and therefore very motivated to work as Web Developer.
              </p>
            </div>
          </div>
        </div>
      </section>
    </div>
    <section
      className="col-md-12 col-lg-4 personal ui-block-color01"
    >
      <h3>
        Personal Info
      </h3>
      <div
        className="personal-info"
      >
        <div
          className="row"
        >
          <div
            className="col-lg-5 col-sm-12"
          >
            Name:
          </div>
          <div
            className="col-lg-7 col-sm-12"
          >
            Michael Kohler
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-lg-5 col-sm-12"
          >
            Date of Birth:
          </div>
          <div
            className="col-lg-7 col-sm-12"
          >
            30/11/1991
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-lg-5 col-sm-12"
          >
            Place:
          </div>
          <div
            className="col-lg-7 col-sm-12"
          >
            Berlin, Germany
            <br />
            originally from Bern, Switzerland
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-lg-5 col-sm-12"
          >
            eMail:
          </div>
          <div
            className="col-lg-7 col-sm-12"
          >
            <a
              href="mailto:me@michaelkohler.info"
            >
              me@michaelkohler.info
            </a>
          </div>
        </div>
        <div
          className="row"
        >
          <div
            className="col-lg-5 col-sm-12"
          >
            Languages (fluent):
          </div>
          <div
            className="col-lg-7 col-sm-12"
          >
            German, English
          </div>
        </div>
      </div>
    </section>
  </section>
  <section
    className="row ui-block-color02"
  >
    <div
      className="col-md-12 col-lg-4 pro-experience"
    >
      <h3>
        Experience
      </h3>
      <div
        className="experience"
      >
        <ul
          className=""
        >
          <li
            className="date"
          >
            04/2008 - now
          </li>
          <li
            className="company"
          >
            <a
              href="https://www.mozilla.org"
            >
              Volunteer @ Mozilla Foundation
            </a>
          </li>
          <li
            className="position"
          >
            Developer, Community Builder
          </li>
          <li>
            Technologies: Node.js, React, Vanilla JavaScript, Rust
          </li>
        </ul>
        <ul
          className=""
        >
          <li
            className="date"
          >
            02/2019 - now
          </li>
          <li
            className="company"
          >
            <a
              href="http://www.control4.com"
            >
              Control4 + SnapAV (acquired NEEO)
            </a>
          </li>
          <li
            className="position"
          >
            Senior Software Developer
          </li>
          <li>
            Technologies: Node.js, React, Lua, Python, Bash
          </li>
        </ul>
        <ul
          className=""
        >
          <li
            className="date"
          >
            11/2016 - 01/2019
          </li>
          <li
            className="company"
          >
            <a
              href="http://www.neeo.com"
            >
              NEEO AG
            </a>
          </li>
          <li
            className="position"
          >
            JavaScript Developer
          </li>
          <li>
            Technologies: Node.js, AngularJS, Ionic, Z-Wave, Bash
          </li>
        </ul>
        <ul
          className=""
        >
          <li
            className="date"
          >
            04/2015 - 10/2016
          </li>
          <li
            className="company"
          >
            <a
              href="http://www.netcase.ch"
            >
              NETCASE GmbH
            </a>
          </li>
          <li
            className="position"
          >
            Web Developer
          </li>
          <li>
            Technologies: Node.js, AngularJS, Bootstrap
          </li>
        </ul>
        <ul
          className=""
        >
          <li
            className="date"
          >
            01/2014 - 02/2015
          </li>
          <li
            className="company"
          >
            <a
              href="http://www.swisscom.ch"
            >
              Swisscom Banking Provider AG
            </a>
          </li>
          <li
            className="position"
          >
            Application Developer, 2nd-Level-Support Document Output Management, Requirement Engineering
          </li>
          <li>
            Technologies: C#, XML, VBA, ITIL
          </li>
        </ul>
        <ul
          className=""
        >
          <li
            className="date"
          >
            08/2010 - 12/2013
          </li>
          <li
            className="company"
          >
            <a
              href="http://www.entris-banking.ch"
            >
              Entris Banking AG
            </a>
          </li>
          <li
            className="position"
          >
            Application Developer, 2nd-Level-Support Document Output Management, Requirement Engineering
          </li>
          <li>
            Technologies: C#, XML, VBA, ITIL
          </li>
        </ul>
      </div>
      <h3>
        Education
      </h3>
      <div
        className="experience"
      >
        <ul
          className=""
        >
          <li
            className="date"
          >
            02/2013 - 02/2017
          </li>
          <li
            className="company"
          >
            <a
              href="https://www.ffhs.ch"
            >
              Fernfachhochschule Schweiz
            </a>
          </li>
          <li
            className="position"
          >
            Computer Science, not finished
          </li>
        </ul>
        <ul
          className=""
        >
          <li
            className="date"
          >
            09/2011 - 01/2013
          </li>
          <li
            className="company"
          >
            <a
              href="http://www.inf.unibe.ch/index_eng.html"
            >
              University of Bern
            </a>
          </li>
          <li
            className="position"
          >
            Computer Science, not finished
          </li>
        </ul>
        <ul
          className=""
        >
          <li
            className="date"
          >
            08/2006 - 07/2010
          </li>
          <li
            className="company"
          >
            <a
              href="http://www.gymneufeld.ch"
            >
              Gymnasium Neufeld
            </a>
          </li>
          <li
            className="position"
          >
            Matura
          </li>
        </ul>
      </div>
    </div>
    <div
      className="col-md-12 col-lg-8 skills ui-block-color03"
    >
      <h3>
        My Professional skills
      </h3>
      <div
        className="row"
      >
        <div
          className="col-md-4 col-sm-12 single-skill"
        >
          <h4>
            Backend JavaScript
          </h4>
          <p>
            I've been developing Node.js applications since 2013, as side projects as well as professionally. Currently at Control4 I'm developing Node.js applications to support the Neeo Remote. Before being acquired by Control4, at NEEO I developed JavaScript applications for both front and back end based with Ionic and Node.js in an IoT environment. Previously during my time at NETCASE GmbH I have also worked on Sails.js and Loopback based APIs.
          </p>
        </div>
        <div
          className="col-md-4 col-sm-12 single-skill"
        >
          <h4>
            Frontend JavaScript
          </h4>
          <p>
            Recently I've mostly used React for my personal as well as Mozilla projects. As part of my work at NEEO and NETCASE GmbH and several private projects I used AngularJS 1 for about 4 years.
          </p>
        </div>
        <div
          className="col-md-4 col-sm-12 single-skill"
        >
          <h4>
            Other languages
          </h4>
          <p>
            Recently I've been doing smaller projects with Lua and Rust as well. Additionally I have also worked with TypeScript for the NEEO SDK. I haven't written any PHP, Ruby or Java in years, but back in the days...
          </p>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-md-4 col-sm-12 single-skill"
        >
          <h4>
            Web Standards
          </h4>
          <p>
            Due to my affiliation with Mozilla, I'm really passionate about the Open Web. Therefore I'm trying to keep on top of changes to Web Standards and will always encourage the use of modern standards where possible.
          </p>
        </div>
        <div
          className="col-md-4 col-sm-12 single-skill"
        >
          <h4>
            DevOps
          </h4>
          <p>
            At NEEO and Control4 I gained experience in DevOps methodologies. I try to make sure Infrastructure as Code is applied where possible. I've helped creating a complete test cluster to test our commits for Control4 in different levels of detail and technology. I've also implemented different metrics tracking systems and visualizations through Grafana. For my private projects I'm using Kubernetes and monitor the health of the cluster as well as my personal Raspberry Pi. I've been using Ansible for configuration management. I test and deploy my personal projects through GitHub Actions, keeping dependencies up to date with dependabot.
          </p>
        </div>
        <div
          className="col-md-4 col-sm-12 single-skill"
        >
          <h4>
            Databases
          </h4>
          <p>
            Nowadays I'me mostly using MySQL for my database needs. In the past, for example at NETCASE, I was also using MongoDB.
          </p>
        </div>
      </div>
    </div>
  </section>
</section>
`;
